plugins/woocommerce-blocks/assets/js/blocks/product-collection/utils.tsx

```diff
@@ -109,3 +119,77 @@ export const addProductCollectionBlockToParentOfPaginationBlock = () => {
 		);
 	}
 };
+
+export const useSetPreviewState = ( {
+	setPreviewState,
+	location,
+	attributes,
+	setAttributes,
+}: {
+	setPreviewState?: SetPreviewState | undefined;
+	location: WooCommerceBlockLocation;
+	attributes: ProductCollectionAttributes;
+	setAttributes: (
+		attributes: Partial< ProductCollectionAttributes >
+	) => void;
+} ) => {
+	const setState = ( newPreviewState: PreviewState ) => {
+		setAttributes( {
+			previewState: {
+				...attributes.previewState,
+				...newPreviewState,
+			},
+		} );
+	};
+
+	// Running setPreviewState function provided by Collection, if it exists.
+	useLayoutEffect( () => {
+		if ( ! setPreviewState ) {
+			return;
+		}
+
+		const cleanup = setPreviewState?.( {
+			setState,
+			location,
+			attributes,
+		} );
+
+		if ( cleanup ) {
+			return cleanup;
+		}
+		// We want this to run only once, adding deps will cause performance issues.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
+
+	/**
+	 * For all Product Collection blocks that inherit query from the template,
+	 * we want to show a preview message in the editor if the block is in
+	 * generic archive template i.e.
+	 * - Products by category
+	 * - Products by tag
+	 * - Products by attribute
+	 */
+	useLayoutEffect( () => {
+		if ( ! setPreviewState ) {
+			const isGenericArchiveTemplate =
+				location.type === LocationType.Archive &&
+				location.sourceData?.termId === null;
+			if ( isGenericArchiveTemplate ) {
+				setAttributes( {
+					previewState: {
+						isPreview: !! attributes?.query?.inherit,
+						previewMessage:
+							'Actual products will vary depending on the page being viewed.',
+					},
+				} );
+			}
+		}
+
+		// We want this to run only once, adding deps will cause performance issues.
```

Comment:
This comment is copy pasta and incorrect, what is your reason for omitting the deps you have omitted?

Reply:
> This comment is copy pasta it doesn't run once, what is your reason for omitting the deps you have omitted? I would prefer to see that in the comment

Ah, Good catch! Actually, I added these [three dependencies](https://github.com/woocommerce/woocommerce/blob/5cfb2acfb6a91e325324b52140096df4c692ec4e/plugins/woocommerce-blocks/assets/js/blocks/product-collection/utils.tsx#L191-L193) in [this commit](https://github.com/woocommerce/woocommerce/commit/d4d0e08e3f83915723019cc27a6a0ce36c185e74) but forgot to update the comment itself 🤦‍♂️ 

Also, I think, in this case, we can add other remaining dependencies, too. Made the changes in commit 7802a6d639c913de15e4a0a13ff8b1c20532ae9d 🤝